### PR TITLE
Feat(front): realization page with css

### PR DESCRIPTION
**Pull request by tomyo002**

### CODE REVIEW

File: frontend/src/style.css [hunk 1]
```diff
@@ -2,3 +2,17 @@
...
+}
+body {
+    display: flex;
+    flex:1;
+}
+#root{
+    display: flex;
+    flex: 1;
+    flex-direction: column;
+}
```
Comment (darkweak) [inline frontend/src/style.css]: Missing NL.

**Review by darkweak — CHANGES_REQUESTED** [review]
When I asked about screenshots, it was in the PR description, not in the code/repository.

**Comment by tomyo002** [comment]
Capture d’écran du 2024-04-19 14-23-40
Capture d’écran du 2024-04-19 14-24-03

File: frontend/src/components/layout/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        className="fill-current"
+        clipRule="evenodd"
+        fillRule="evenodd"
+        height="36"
+        viewBox="0 0 24 24"
+        width="36"
+        xmlns="http://www.w3.org/2000/svg"
+      >
+        <path d="M22.672 15.226l-2.432.811.841 2.515c.33 1.019-.209 2.127-1.23 2.456-1.15.325-2.148-.321-2.463-1.226l-.84-2.518-5.013 1.677.84 2.517c.391 1.203-.434 2.542-1.831 2.542-.88 0-1.601-.564-1.86-1.314l-.842-2.516-2.431.809c-1.135.328-2.145-.317-2.463-1.229-.329-1.018.211-2.127 1.231-2.456l2.432-.809-1.621-4.823-2.432.808c-1.355.384-2.558-.59-2.558-1.839 0-.817.509-1.582 1.327-1.846l2.433-.809-.842-2.515c-.33-1.02.211-2.129 1.232-2.458 1.02-.329 2.13.209 2.461 1.229l.842 2.515 5.011-1.677-.839-2.517c-.403-1.238.484-2.553 1.843-2.553.819 0 1.585.509 1.85 1.326l.841 2.517 2.431-.81c1.02-.33 2.131.211 2.461 1.229.332 1.018-.21 2.126-1.23 2.456l-2.433.809 1.622 4.823 2.433-.809c1.242-.401 2.557.484 2.557 1.838 0 .819-.51 1.583-1.328 1.847m-8.992-6.428l-5.01 1.675 1.619 4.828 5.011-1.674-1.62-4.829z"></path>
+      </svg>
```
Comment (darkweak) [inline frontend/src/components/layout/Footer.tsx]: Useless SVG right?

File: frontend/src/components/layout/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+import { Connected } from '../molecule';
+
+type PropsNavbar = {
```
Comment (darkweak) [inline frontend/src/components/layout/Navbar.tsx]: Naming.

File: frontend/src/components/molecule/Connected.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React, { useEffect, useState } from 'react';
+
+import { extractor } from '../../service/fetcher';
+import { AllowedIcon, Icon } from '../atomic';
+
+type PropsConnected = {
```
Comment (darkweak) [inline frontend/src/components/molecule/Connected.tsx]: Naming.

File: frontend/src/components/molecule/DisplayQuantile.tsx [hunk 5]
```diff
@@ -42,24 +42,26 @@ export const DisplayQuantile: React.FC<DisplayProps> = ({
           });
         } else {
           const index = lines.findIndex(line => line['name'] === name);
-          const tmpPath = lines[index]['metrics'][0];
+          const path = lines[index].metrics[0];
```
Comment (darkweak) [inline frontend/src/components/molecule/DisplayQuantile.tsx]: You're accessing to one `metric` so why not naming your var `metric`?

File: frontend/src/components/molecule/DisplayQuantile.tsx [hunk 6]
```diff
@@ -22,7 +22,7 @@ export const DisplayQuantile: React.FC<DisplayProps> = ({
   const [dataSum, setDataSum] = useState<ReadonlyArray<number>>([]);
   const [label, setlabel] = useState<Array<string>>([]);
   const [labelSum, setlabelSum] = useState<Array<string>>([]);
-  const InitializeClick = () => {
+  const ResetClick = () => {
```
Comment (darkweak) [inline frontend/src/components/molecule/DisplayQuantile.tsx]: Do you need this constant?

File: frontend/src/components/page/DisplayPage.tsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+          title="process resident memory bytes"
+        />
+        <Display
+          baseUrl={baseUrl}
+          name="process_virtual_memory_bytes"
+          timeMilliSecond={1000}
+          title="process virtual memory bytes"
+        />
+      </div>
+      <div className="flex flex-col ">
```
Comment (darkweak) [inline frontend/src/components/page/DisplayPage.tsx]:
```suggestion
      <div className="flex flex-col">
```
By the way, `flex flex-col` with only one item?

File: frontend/src/components/page/DisplayPage.tsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+import { Navbar, Footer } from '../layout';
+import { Display, DisplayQuantile } from '../molecule';
+
+type PropsDisplayPage = {
```
Comment (darkweak) [inline frontend/src/components/page/DisplayPage.tsx]: Naming.

File: frontend/tailwind.config.js [hunk 9]
```diff
@@ -4,4 +4,7 @@ export default {
     extend: {},
   },
   plugins: [require('@tailwindcss/typography'), require('daisyui')],
+  daisyui: {
+    themes: ['light', 'coffee'],
```
Comment (darkweak) [inline frontend/tailwind.config.js:8]: `coffee`? Really?

File: main.go [hunk 10]
```diff
@@ -19,7 +19,7 @@ func main() {
 	err := wails.Run(&options.App{
 		Title:  "Souin Desktop",
 		Width:  1024,
-		Height: 768,
+		Height: 853,
```
Comment (darkweak) [inline main.go]: Arbitrary height?

**Review by darkweak — CHANGES_REQUESTED** [review]
Take the time to apply all previous comments in your previous PRs

**Comment by darkweak** [comment]
Here is what it should look like:
image

I removed the theme switch and made a multiline graph so it can be reused easier.

File: frontend/src/components/molecule/MultiLineData.tsx [hunk 11]
```diff
@@ -29,8 +35,12 @@ export const Display: React.FC<DisplayProps> = ({ baseUrl, labels, title }) => {
         if (!lines) {
           setMultiData(current => {
             return current.map(data => {
+              const currentData =
+                data['data'].length >= max
+                  ? data['data'].slice(1)
+                  : data['data'];
```
Comment (darkweak) [inline frontend/src/components/molecule/MultiLineData.tsx]:
```diff
            return current.map(({ data }) => {
              const currentData =
                data.length >= max
                  ? data.slice(1)
                  : data;
```

File: frontend/src/service/fetcher.tsx [hunk 12]
```diff
@@ -19,3 +19,13 @@ export function extractor(baseUrl: string, endpoint: string) {
       console.warn(error);
     });
 }
+
+export function checkHealth(baseUrl: string) {
+  return fetcher(baseUrl, '/')
+    .then(() => {
+      return 'ok';
```
Comment (darkweak) [inline frontend/src/service/fetcher.tsx]: Plaintext `'ok'`? Why?

File: frontend/src/components/molecule/HealthCheck.tsx [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+type HealthCheckProps = {
+  baseUrl: string;
+};
+
+export const HealthCheck: React.FC<HealthCheckProps> = ({ baseUrl }) => {
+  const [icon, seticon] = useState<AllowedIcon>('x-circle');
+  useEffect(() => {
+    const interval = setInterval(() => {
+      checkHealth(baseUrl).then(response => {
+        if (response === null) {
```
Comment (darkweak) [inline frontend/src/components/molecule/HealthCheck.tsx]: Simplify this check.

File: frontend/src/components/molecule/MultiLineData.tsx [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+              const currentData =
+                data.data.length >= max ? data.data.slice(1) : data.data;
+              return {
+                data: [...currentData, 0],
+                label: data.label,
+                borderColor: data.borderColor,
+              };
+            });
+          });
+        } else {
```
Comment (darkweak) [inline frontend/src/components/molecule/MultiLineData.tsx]: duplicated code.

File: frontend/src/components/molecule/MultiLineData.tsx [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+
+export const MultiLineData: React.FC<MultiLineDataProps> = ({
+  baseUrl,
+  labels,
+  title,
+  max,
+}) => {
+  const [multiData, setMultiData] = useState<ReadonlyArray<MultiDataTypes>>(
+    labels.map(label => {
+      return {
```
Comment (darkweak) [inline frontend/src/components/molecule/MultiLineData.tsx]: `return` not needed.

File: frontend/src/components/molecule/MultiLineData.tsx [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+      extractor(baseUrl, '/metrics').then(lines => {
+        if (!lines) {
+          setMultiData(current => {
+            return current.map(data => {
+              const currentData =
+                data.data.length >= max ? data.data.slice(1) : data.data;
+              return {
+                data: [...currentData, 0],
+                label: data.label,
+                borderColor: data.borderColor,
```
Comment (darkweak) [inline frontend/src/components/molecule/MultiLineData.tsx]: Simplify.

File: frontend/src/components/molecule/MultiLineData.tsx [hunk 17]
```diff
@@ -0,0 +1,90 @@
...
+              };
+            });
+          });
+        }
+      });
+
+      setlabel(current => {
+        const currentLabel = current.length >= max ? current.slice(1) : current;
```
Comment (darkweak) [inline frontend/src/components/molecule/MultiLineData.tsx]: simplify.

File: frontend/src/components/molecule/MultiLineData.tsx [hunk 18]
```diff
@@ -0,0 +1,90 @@
...
+  );
+  const [label, setlabel] = useState<Array<string>>([]);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      extractor(baseUrl, '/metrics').then(lines => {
+        if (!lines) {
+          setMultiData(current => {
+            return current.map(data => {
+              const currentData =
```
Comment (darkweak) [inline frontend/src/components/molecule/MultiLineData.tsx]: Useless assignment.

File: frontend/src/components/layout/Footer.tsx [hunk 19]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Comment (darkweak) [inline frontend/src/components/layout/Footer.tsx]: Rework.

File: frontend/src/components/molecule/Line.tsx [hunk 20]
```diff
@@ -1,27 +1,31 @@
...
 import 'chart.js/auto';
 import { Line as LineJs } from 'react-chartjs-2';
+import { MultiDataTypes } from 'src/utils';
 
 type LineProps = {
-  data: ReadonlyArray<number>;
-  title: string;
-  label: Array<string>;
+  data: ReadonlyArray<MultiDataTypes>;
+  labels: Array<string>;
```
Comment (darkweak) [inline frontend/src/components/molecule/Line.tsx]: `ReadonlyArray`

**Review by darkweak — CHANGES_REQUESTED** [review]

File: frontend/src/components/molecule/MultiLineData.tsx [hunk 21]
```diff
@@ -18,56 +18,39 @@ export const MultiLineData: React.FC<MultiLineDataProps> = ({
...
-                data.data.length >= max ? data.data.slice(1) : data.data;
-              return {
-                data: [...currentData, 0],
-                label: data.label,
-                borderColor: data.borderColor,
-              };
-            });
+        setMultiData(currents => {
+          return currents.map(({ data, label, ...rest }) => {
+            return {
```
Comment (darkweak) [inline frontend/src/components/molecule/MultiLineData.tsx]: `return` to needed.

File: frontend/src/components/molecule/MultiLineData.tsx [hunk 22]
```diff
@@ -18,56 +18,39 @@ export const MultiLineData: React.FC<MultiLineDataProps> = ({
...
-              const currentData =
-                data.data.length >= max ? data.data.slice(1) : data.data;
-              return {
-                data: [...currentData, 0],
-                label: data.label,
-                borderColor: data.borderColor,
-              };
-            });
+        setMultiData(currents => {
+          return currents.map(({ data, label, ...rest }) => {
```
Comment (darkweak) [inline frontend/src/components/molecule/MultiLineData.tsx]: `return` to needed.

File: frontend/src/components/molecule/MultiLineData.tsx [hunk 23]
```diff
@@ -18,56 +18,39 @@ export const MultiLineData: React.FC<MultiLineDataProps> = ({
...
+              data: [
+                ...(data.length >= max ? data.slice(1) : data),
+                !lines
+                  ? 0
+                  : parseFloat(
+                      lines[lines.findIndex(line => line.name === label)]
+                        .metrics[0].value,
+                    ),
+              ],
+              label: label,
```
Comment (darkweak) [inline frontend/src/components/molecule/MultiLineData.tsx]: To simplify.

File: frontend/src/service/fetcher.tsx [hunk 24]
```diff
@@ -21,11 +21,7 @@ export function extractor(baseUrl: string, endpoint: string) {
...
-  return fetcher(baseUrl, '/')
-    .then(() => {
-      return 'ok';
-    })
-    .catch(() => {
-      return null;
-    });
+  return fetcher(baseUrl, '/').catch(() => {
+    return null;
+  });
```
Comment (darkweak) [inline frontend/src/service/fetcher.tsx]: return a boolean.

File: frontend/src/components/layout/Footer.tsx [hunk 25]
```diff
@@ -0,0 +1,5 @@
+export const Footer: React.FC = () => (
+  <footer className="footer-center p-4 bg-neutral text-neutral-content">
```
Comment (darkweak) [inline frontend/src/components/layout/Footer.tsx]: `footer` className is missing.

**Review by darkweak — CHANGES_REQUESTED** [review]

File: frontend/src/components/layout/Navbar.tsx [hunk 26]
```diff
@@ -0,0 +1,21 @@
...
+  name: string;
+  baseUrl: string;
+};
+
+export const Navbar: React.FC<NavbarProps> = ({ name, baseUrl }) => (
+  <div className="navbar bg-neutral text-neutral-content">
+    <div className="flex-1">
+      <a className="btn btn-ghost text-xl">
+        {name}: {baseUrl}
+      </a>
```
Comment (darkweak) [inline frontend/src/components/layout/Navbar.tsx]: Use span instead.

File: frontend/src/components/layout/Navbar.tsx [hunk 27]
```diff
@@ -0,0 +1,21 @@
...
+    </div>
+    <div className="flex-none">
```
Comment (darkweak) [inline frontend/src/components/layout/Navbar.tsx]: Not needed.

File: frontend/src/components/molecule/HealthCheck.tsx [hunk 28]
```diff
@@ -0,0 +1,23 @@
...
+};
+
+export const HealthCheck: React.FC<HealthCheckProps> = ({ baseUrl }) => {
+  const [icon, setIcon] = useState<AllowedIcon>('x-circle');
+  useEffect(() => {
+    const interval = setInterval(() => {
+      checkHealth(baseUrl).then(response => {
+        setIcon(!response ? 'x-circle' : 'check-circle');
+      });
+    }, 1000);
```
Comment (darkweak) [inline frontend/src/components/molecule/HealthCheck.tsx]: Remove the interval, check only once.

**Review by darkweak — APPROVED** [review]

**Pull request merged**